lib/manager/npm/extract/pnpm.ts

```diff
@@ -102,7 +102,11 @@ export async function detectPnpmWorkspaces(
     }
     const packageFilters = packageFilterCache.get(workspaceYamlPath);
     const isPackageInWorkspace =
-      packageFilters !== null && matchesAnyPattern(packageFile, packageFilters);
+      packageFilters !== null &&
+      matchesAnyPattern(
+        packageFile,
+        packageFilters.map((filter) => `${filter}/*`)
```

Comment:
I have a concern that adding `/*` to existing patterns which already end in a pattern will cause the original pattern to break

Reply:
I wasn't that happy with this myself, so I finally figured out how pnpm package matching differs from renovates.

https://github.com/pnpm/pnpm/blob/a0649310a2f692d9320b947f30d059ea5ca86e65/packages/find-packages/src/index.ts#L71-L87

I think doing it exactly like that (aaea299bae9cdbdfb7a088ab6b9fac8da6fb2059) should be fine.